migrations/2001.changefeed-v2.sql

```diff
@@ -395,30 +398,33 @@ as begin
             @shard_id = @shard_id,
             @lock_timeout = -1,
             @lock_result = @lock_result output;
-        if @lock_result = 1
-        begin
-            rollback
-            -- 1 means "got lock after timeout". This means someone else has processed the outbox;
-            -- so we try again once. If there are no results here, it just means the outbox was empty
-            -- just now anyway .. the caller should sleep a bit before retrying.
-
-            insert into #read(ulid, ', @pklist,')
-            select top(@pagesize)
-                ulid,
-                ', @pklist, '
-            from ', @feed_table, '
-            where
-                shard_id = @shard_id
-                and ulid > @cursor;
-            return
-        end;
 
         if @lock_result < 0
         begin
             throw 77100, ''Error getting lock'', 1;
         end;
 
-        -- @lock_result = 0; got lock without waiting. Consume outbox.
+        -- At this point it does not matter if we got the lock without waiting or not, in BOTH
```

Comment:
Issue 2: Used to re-check feed only if we got a timeout on the lock. But the timeout on the lock doesn't really mean much, we need to always re-check